platform/fabric/core/generic/committer.go

```diff
@@ -298,29 +298,39 @@ func (c *channel) postProcessTx(txid string) error {
 // SubscribeTxStatusChanges registers a listener for transaction status changes for the passed transaction id.
 // If the transaction id is empty, the listener will be called for all transactions.
 func (c *channel) SubscribeTxStatusChanges(txID string, listener driver.TxStatusChangeListener) error {
-	topic := compose.CreateCompositeKeyOrPanic(&strings.Builder{}, "tx", c.network.Name(), c.name, txID)
+	var topic string
+	if len(txID) == 0 {
+		topic = compose.CreateCompositeKeyOrPanic(&strings.Builder{}, "tx", c.network.Name(), c.name)
+	} else {
+		topic = compose.CreateCompositeKeyOrPanic(&strings.Builder{}, "tx", c.network.Name(), c.name, txID)
+	}
```

Comment:
I suggest to move this into a small helper function `createTopic` and use it instead of repeating the code